internal/server/data/group.go

```diff
@@ -42,7 +42,45 @@ func DeleteGroups(db *gorm.DB, selectors ...SelectorFunc) error {
 		if err != nil {
 			return err
 		}
+
+		identities, err := ListIdentitiesByGroup(db, g.ID, []SelectorFunc{}...)
+		if err != nil {
+			return err
+		}
+
+		err = RemoveUsersFromGroup(db, g.ID, identities)
+		if err != nil {
+			return err
+		}
 	}
 
 	return deleteAll[models.Group](db, ByIDs(ids))
 }
+
+func AddUsersToGroup(db *gorm.DB, groupID uid.ID, identities []models.Identity) error {
+	for _, id := range identities {
+		var ids []uid.ID
+
+		if err := db.Raw("SELECT identity_id FROM identities_groups WHERE identity_id = ? AND group_id = ?", id.ID, groupID).Scan(&ids).Error; err != nil {
+			return err
+		}
+
+		if len(ids) == 0 {
+			err := db.Exec("INSERT INTO identities_groups (identity_id, group_id) VALUES (?, ?)", id.ID, groupID).Error
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
```

Comment:
Are you thinking of a multi-row insert statement? i.e.:

```
    err := db.Exec(fmt.Sprintf("INSERT INTO identities_groups (identity_id, group_id) VALUES %s", idPairs)).Error
```
We can definitely do that to cut down on the number of insert statements, but we still need to weed out duplicate
IDs, so you'll still end up calling select a pile of times. Otherwise if you try to insert an ID pair which already exists the unique index will prevent the entire insert statement from completing.

Reply:
I ended up keeping the loop but simplifying the insert to include a select statement to catch if the row already exists in the groups/identity membership table. The result should be safe across both sqlite/postgres. We can further optimize in the future if we need to.